docker/u1604/Dockerfile

```diff
@@ -1,6 +1,8 @@
 FROM openmicroscopy/ome-files-cpp-u1604
 MAINTAINER ome-devel@lists.openmicroscopy.org.uk
 
+RUN apt-get update
```

Comment:
This will lead to some bloat in the image. Is this still necessary after `docker pull openmicroscopy/ome-files-cpp-u1604`? Alternatively, is there anything we can do in the upstream image to avoid this?

Reply:
It's always necessary.  The cache gets out of date, and then package fetching fails.  Unless you're recreating the upstream image daily, then this is needed to avoid failure.

Reply:
Why is it only occurring now?

Reply:
It's the first PR opened in several months using this image, so is failing as a consequence.  `apt` is no different to `brew` or `yum` in requiring an up-to-date cache to guarantee successful operation.  The only difference is that the latter two update their cache automatically.  Usually you can get away with it for a few weeks since the old packages are retained for some time for this reason, but in this case it has been well over 4 months since the last PR, and potentially even longer since the image was updated.

There won't be any bloat from doing an update.  It's replacing the existing package list with an updated list; there's no extra space usage other than a trivial amount from any newly added packages.

Reply:
> apt is no different to brew or yum in requiring an up-to-date cache to guarantee successful operation.

That should be happening in the parent image though: they should be periodically rebuilt.

> There won't be any bloat from doing an update. It's replacing the existing package list with an updated list; there's no extra space usage other than a trivial amount from any newly added packages.

Fair enough, but it is creating a new layer and invalidating others.

Reply:
At the moment, I don't think that is happening, so this is needed to make this work.  OTOH, using docker here with a hardcoded set of upstream dependencies is quite inflexible and replacing it with `ome-files-build` and/or gitlab would be desirable to have an effective CI build.  In the interim, this keeps what we have working.